auction-app-frontend/src/components/CheckoutComponent/CheckoutComponent.jsx

```diff
@@ -0,0 +1,124 @@
+import React, { useState, useEffect } from "react";
+import { Elements } from "@stripe/react-stripe-js";
+import { loadStripe } from "@stripe/stripe-js";
+import { useNavigate } from "react-router-dom";
+
+import { CheckoutAddressForm, CheckoutPaymentForm, LoadingComponent, ErrorComponent } from "src/components";
+import { STRIPE_PUBLIC_KEY, CHECKOUT_STEPS, ROUTE_PATHS } from "src/constants";
+import { createPaymentIntent } from "src/services";
+import { addPaymentInfo } from "src/services/paymentService";
+import { useUser } from "src/store/UserContext";
+import { getUser } from "src/services/userService";
+
+import "./style.scss";
+
+const stripePromise = loadStripe(STRIPE_PUBLIC_KEY);
```

Comment:
Why is this outside the component?

Reply:
to avoid reinitializing stripe each time this component rerenders